### DISCUSSION TITLE
abi.map is not a function UseContractRead

### DISCUSSION THREAD

**Discussion by dev-cz** [wevm/wagmi#703] [Q&A] [answered]
Getting the below error. According to docs, I have all the right info.

I exported the json file of the contract and imported it.

![image](https://user-images.githubusercontent.com/51761060/178852444-b6165fb7-6ec0-45ab-b276-f9c6258cb3e5.png)

This is  the code I am using, and in my return piece just have the data.
![image](https://user-images.githubusercontent.com/51761060/178852647-6c4e0db2-742e-4420-a3ee-9c533a84dfa8.png)

And then calling the data on my page.
![image](https://user-images.githubusercontent.com/51761060/178852692-683abc08-21ae-4048-9290-03559c8cc96c.png)

**Comment by fdarian** [comment]
Try removing the cache on Dev Tool -> Application -> Local Storage -> `your-url` -> "wagmi-cache" key.

I think this should be converted into an issue, I also had this issue and removing cache shouldn't be the solution.

**Reply by dev-cz** [reply]
Thanks, not sure what it was. But I think it was the way the ABI is constructed from etherscan when exporting.

I ended up resolving it by pasting the ABI into this site, https://abi.hashex.org/, then selecting "Parse", and using the updated format from that site.

**Reply by fecony** [reply]
Clearing the cache helped... God, why did I waste a few hours for this 😭

**Comment by dev-cz** [comment, marked as answer]
I ended up resolving it by pasting the ABI into this site, https://abi.hashex.org/, then selecting "Parse", and using the updated format from that site.
